pkg/util/httpproxy/upgradeawareproxy.go

```diff
@@ -172,6 +188,7 @@ func (p *UpgradeAwareSingleHostReverseProxy) serveUpgrade(w http.ResponseWriter,
 	done := make(chan struct{}, 2)
 
 	go func() {
+		// TODO: strip any WWW-Authenticate challenge headers if the initial response is a 401
```

Comment:
This comment belongs in the 2nd copy goroutine (io.Copy(requestHijackedConn, backendConn))